### ISSUE TITLE
# [FEAT][kernels]: fused logp CUDA kernels lack backward functions

### ISSUE THREAD

**Issue by KJLdefeated** [Flink-ddd/RL-Engine#100]
## Problem

The fused logp CUDA kernels are **forward-only**. Every symbol exported from the extension is a forward variant (`csrc/ops.cpp`):

```
fused_logp, fused_logp_sm90,
fused_logp_forward_{out, fp32, indexed_out, indexed_fp32, online_out, online_fp32, online_indexed_out, online_indexed_fp32}
```
Neither kernel file implements a gradient:
- `csrc/fused_logp_kernel.cu` — generic fused logp (all RL variants)
- `csrc/cuda/fused_logp_sm90.cu` — TMA-accelerated SM90 path

## Why it matters

`logp(logits, token_ids)` is differentiable math: in RL training the policy logits are the live network output, and the loss must backprop through the selected log-probs.

## Proposed solution
```
d logp[a] / d logits[v] = 1[v == a] - p[v]
grad_logits[v] = grad_out * (1[v == a] - p[v])
```

## Affected files

- `csrc/fused_logp_kernel.cu` — add backward kernel (covers all forward variants)
- `csrc/cuda/fused_logp_sm90.cu` — add backward for the TMA path
- `csrc/ops.cpp` — bind the backward
- `rl_engine/kernels/ops/cuda/loss/logp.py` — wrap in `torch.autograd.Function`
- `tests/` — gradient tests vs the native reference

**Comment by Dnoob** [comment]
Hi，I'd like to take this issue

**Comment by Flink-ddd** [comment]
cc @KJLdefeated , PTAL

**Comment by KJLdefeated** [comment]
@Dnoob  Sure!
